ls1intum/Artemis
# [Refactor] the existing Modeling Exercise tests to work with pageobjects

### Motivation and Context
 Pageobjects make our tests more easily maintainable. 

### Description

I have created pageobjects for various pages involved in the creation of modeling exercises and involve these in the tests.
I also made changes to the api requests used in the tests.

### Steps for Testing
 

Look at code and to run locally:
Pull branch
In cypress.json: 
- set baseUrl to testserver of your choosing (ONLY TS3 because you need admin access)

In cypress.env.json:
- set "username" and "password" to the one used <a href="https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=25252245">here</a> except put "USERID" for the variable part.
- set "adminUsername" and "adminPassword" to an admin account you have access to (probably your own rbg account)

$ yarn run cypress open
run ModelingExerciseSpec or ideally all specs to make sure everything works
alternatively to run headlessly: $ yarn run cypress run --spec "src/test/cypress/integration/ModelingExercise.spec.ts"
You can also configure all of these options to test your locally hosted version of artemis, but i dont recommend it.
